### PR TITLE
Implement gRPC for `echo` and `version` APIs

### PR DESCRIPTION
**Pull request by JayKaku**
@stefanprodan here is the restructured api  to api/http and api/grpc along with adding functionality of version and echo api

### CODE REVIEW

**Comment by Rachitmehrotra** [comment]
@stefanprodan please review this.

**Comment by JayKaku** [comment]
@stefanprodan added logger and config to ehco

File: pkg/api/grpc/echo.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+)
+
+type echoServer struct {
+	echo.UnimplementedEchoServiceServer
+	config *Config
+	logger *zap.Logger
+}
+
+func (s *echoServer) Echo (ctx context.Context, message *echo.Message) (*echo.Message, error){
+	log.Printf("Received message body from client: %s", message.Body)
```
Comment (stefanprodan) [inline pkg/api/grpc/echo.go:18]: Please use the zap logger

File: pkg/api/grpc/echo.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import (
+	"context"
+	"log"
+
+	"github.com/stefanprodan/podinfo/pkg/api/grpc/echo"
+	"go.uber.org/zap"
+)
+
+type echoServer struct {
+	echo.UnimplementedEchoServiceServer
```
Comment (stefanprodan) [inline pkg/api/grpc/echo.go:12]: Why is this call Unimplemented?

Comment (JayKaku) [reply pkg/api/grpc/echo.go:11]: That's something protoc-gen-grpc-go compiler now throws an error for in newer version `must be embedded to have forward compatible implementations.`, it act as a placeholder or a default behavior for any methods that have not yet been implemented in your code

**Comment by Rachitmehrotra** [comment]
@stefanprodan please review

**Comment by stefanprodan** [comment]
Tests panic here `/home/runner/work/podinfo/podinfo/pkg/api/grpc/echo.go:18`

**Comment by Prashant-Dwivedi-08-01** [comment]
@stefanprodan can you please review it once again

**Comment by Rachitmehrotra** [comment]
@stefanprodan please review

**Comment by JayKaku** [comment]
@stefanprodan could you please review and let us know? Thanks!

**Comment by Prashant-Dwivedi-08-01** [comment]
@stefanprodan Could you please review it once, so that we can go ahead with next set APIs, it has been a while now. Thanks.

**Comment by Rachitmehrotra** [comment]
@stefanprodan please review this; we want to contribute more .

**Comment by stefanprodan** [comment]
Hey sorry for the delay, I've been busy with work and didn't had time left for podinfo. Can you please rebase with master and force push so the test will run. Thanks for your patience!

**Comment by JayKaku** [comment]
> Hey sorry for the delay, I've been busy with work and didn't had time left for podinfo. Can you please rebase with master and force push so the test will run. Thanks for your patience!

Done! Force pushed the branch after a rebase with the master

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @JayKaku and @Rachitmehrotra 🏅

**Pull request merged**